File: 30 Days of Code/Python/12 - Day 11 - 2D Arrays.py
# ========================
#       Information
# ========================

# Direct Link: https://www.hackerrank.com/challenges/30-2d-arrays/problem
# Difficulty: Easy
# Max Score: 30
# Language: Python

# ========================
#         Solution
# ========================

if __name__ == '__main__':
    ARRAY = []
    ADD = []
    MAX_SUM = []

    for _ in range(6):
        ARRAY.append(list(map(int, input().rstrip().split())))

    for i in range(len(ARRAY)-2):
        for j in range(4):
            ADD.extend(ARRAY[i][j:j+3])
            ADD.append(ARRAY[i+1][j+1])
            ADD.extend(ARRAY[i+2][j:j+3])
            MAX_SUM.append(sum(ADD))
            ADD = []

    print(max(MAX_SUM))

File: 30 Days of Code/C#/12 - Day 11 - 2D Arrays.cs
// ========================
//       Information
// ========================

// Direct Link: https://www.hackerrank.com/challenges/30-2d-arrays/problem
// Difficulty: Easy
// Max Score: 30
// Language: C#

// ========================
//         Solution
// ========================

using System;

class Solution {

    static void Main(string[] args) {
        int[][] arr = new int[6][];

        for (int i = 0; i < 6; i++) {
            arr[i] = Array.ConvertAll(Console.ReadLine().Split(' '), arrTemp => Convert.ToInt32(arrTemp));
        }
        int result = -9 * 7;
        int rowCount = arr.Length;
        int columnCount = arr[0].Length;

        for (int c = 0; c < columnCount - 2; c++) {
            for (int r = 0; r < rowCount - 2; r++) {
                int firstRowSum = arr[r][c] + arr[r][c + 1] + arr[r][c + 2];
                int secondRowSum = arr[r + 1][c + 1];
                int thirdRowSum = arr[r + 2][c] + arr[r + 2][c + 1] + arr[r + 2][c + 2];
                int totalHourGlass = firstRowSum + secondRowSum + thirdRowSum;

                result = Math.Max(result, totalHourGlass);
            }
        }

        Console.WriteLine(result);
    }
}
